marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/main.sjs

```diff
@@ -102,6 +102,11 @@ function main(contentSequence, options, stepExecutionContext) {
           entityName = lib.getEntityName(fn.string(currentRelatedMapping.targetEntityType));
           entityContext = createContextForRelatedEntityInstance(currentRelatedMapping);
         }
+
+        if(content.context && content.context.originalCollections){
```

Comment:
I think this should be done in the createContextForRelatedEntityInstance function, as doesn't it only apply for a related entity mapping?